webview-ui/src/components/account/AccountWelcomeView.tsx

```diff
@@ -1,37 +1,38 @@
-import { VSCodeButton, VSCodeLink } from "@vscode/webview-ui-toolkit/react"
-import { useClineSignIn } from "@/context/ClineAuthContext"
+import { VSCodeLink } from "@vscode/webview-ui-toolkit/react"
+import { useClineAuth } from "@/context/ClineAuthContext"
 import { useExtensionState } from "@/context/ExtensionStateContext"
 import ClineLogoVariable from "../../assets/ClineLogoVariable"
+import { ClineAuthStatus } from "./ClineAuthStatus"
 
-// export const AccountWelcomeView = () => (
-// 	<div className="flex flex-col items-center pr-3 gap-2.5">
-// 		<ClineLogoWhite className="size-16 mb-4" />
 export const AccountWelcomeView = () => {
 	const { environment } = useExtensionState()
-	const { isLoginLoading, handleSignIn } = useClineSignIn()
+	const { hasSessionData } = useClineAuth()
+
+	if (!hasSessionData) {
+		return (
+			<div className="flex flex-col items-center pr-3 gap-2.5">
+				<ClineLogoVariable className="size-16 mb-4" environment={environment} />
+
+				<p>
+					Sign up for an account to get access to the latest models, billing dashboard to view usage and credits, and
+					more upcoming features.
+				</p>
+
+				<ClineAuthStatus authButtonText="Sign up with Cline" />
+
+				<p className="text-(--vscode-descriptionForeground) text-xs text-center m-0">
+					By continuing, you agree to the <VSCodeLink href="https://cline.bot/tos">Terms of Service</VSCodeLink> and{" "}
+					<VSCodeLink href="https://cline.bot/privacy">Privacy Policy.</VSCodeLink>
+				</p>
+			</div>
+		)
+	}
 
 	return (
 		<div className="flex flex-col items-center pr-3 gap-2.5">
 			<ClineLogoVariable className="size-16 mb-4" environment={environment} />
 
-			<p>
-				Sign up for an account to get access to the latest models, billing dashboard to view usage and credits, and more
-				upcoming features.
-			</p>
-
-			<VSCodeButton className="w-full mb-4" disabled={isLoginLoading} onClick={handleSignIn}>
-				Sign up with Cline
-				{isLoginLoading && (
-					<span className="ml-1 animate-spin">
-						<span className="codicon codicon-refresh"></span>
-					</span>
-				)}
-			</VSCodeButton>
-
-			<p className="text-(--vscode-descriptionForeground) text-xs text-center m-0">
-				By continuing, you agree to the <VSCodeLink href="https://cline.bot/tos">Terms of Service</VSCodeLink> and{" "}
-				<VSCodeLink href="https://cline.bot/privacy">Privacy Policy.</VSCodeLink>
-			</p>
+			<ClineAuthStatus authButtonText="Sign up with Cline" />
```

Comment:
In AccountView.tsx, we have set it up to show the AccountWelcomeView if the user is not logged in. If user is logged in, we would show ClineAccountView instead. We should probably move the hasSessionData check to AccountView instead and remove ClineAuthStatus in this component?